[Security Solution][Entity Analytics][PrivMon] Handle null/undefined values gracefully for tiles on Privmon dashboard

abhishekbhatia1710 opened the pull request:
## Summary

What changed:

Added error handling for PrivMon dashboard tiles when ES queries fail (like missing ML indices)
Instead of showing broken visualizations, tiles now display a clean "Data not available" message
The error tiles maintain consistent styling and layout with working tiles

Added state tracking to detect when visualization loading completes
- Check for missing [tables] property in visualization response to identify errors
- Render a styled error state using EUI components that matches the design of working tiles
- Made error message i18n compliant for proper localization

Screenshot Ref : 

![image](https://github.com/user-attachments/assets/3e989159-5fa2-4b57-9fe4-523f5785e109)

With data : 

![image](https://github.com/user-attachments/assets/473f3254-32c2-4799-ba4b-51121932ab20)


Testing Steps : 
1. Checkout this PR
2. Start the privmon engine : `POST kbn:/api/entity_analytics/monitoring/engine/init`
3. Add some privileged users either using a CSV file or from dev console.
4. Add relevant data for different indexes using the security-documents-generator. 
6. Enable the risk score and Asset criticality.
7. Change filters for timerange or any KQL query to get the desired results.
8. The tile visualisations on the privmon dashboard should not display any errors. It should either display the Count or `Data not available`.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.






<!--ONMERGE {"backportTargets":["9.1"]} ONMERGE-->

abhishekbhatia1710 committed:
Changes to show "Data not available" when index is not present for a specific data

elasticmachine commented:
Pinging @elastic/security-entity-analytics (Team:Entity Analytics)

abhishekbhatia1710 committed:
Reset when filter changes

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/315376)
* Commit: 86714b49ec8929a3a7636e93d4f57ec4dac0c7db

### Failed CI Steps
* [Defend Workflows Cypress Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/315376#0197cff1-1f65-4ba9-a23e-875995f56c35)

cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315376","number":315376,"commit":"86714b49ec8929a3a7636e93d4f57ec4dac0c7db"}],"number":315376}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
Merge branch 'main' into privmon-handle-null-values-for-tiles

abhishekbhatia1710 committed:
Adjusted to better min-width

abhishekbhatia1710 committed:
Merge branch 'main' into privmon-handle-null-values-for-tiles

jaredburgettelastic reviewed (approved):
Have **not** desk tested, but code looks straightforward and good to me!

abhishekbhatia1710 commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/common/key_insights_tile.tsx:
```diff
@@ -61,30 +55,82 @@ export const KeyInsightsTile: React.FC<KeyInsightsTileProps> = ({
+  ) {
+    return (
+      <EuiFlexItem grow={false} style={{ minWidth: LENS_VISUALIZATION_MIN_WIDTH }}>
+        <EuiPanel
```
We can discuss if this looks better : 

![image](https://github.com/user-attachments/assets/06782f7d-5990-486f-95d0-2fb3408760df)

hop-dev replied:
Yeah or maybe big central icon or text below, one for 🅿️

abhishekbhatia1710 committed:
Addressing review comments :
1. minWidht and minHeight remains
2. Subdued tile is removed
3. Title and "Data not available" similar to other tiles
4. Warning icon currently near "Data not available"

abhishekbhatia1710 committed:
Adjusting min width for the lens attributes

elasticmachine committed:
Merge branch 'main' into privmon-handle-null-values-for-tiles
